documentation/api/io.strimzi.api.kafka.model.EntityUserOperatorSpec.adoc

```diff
@@ -38,9 +40,13 @@ spec:
         type: inline
         loggers:
           rootLogger.level: INFO
+          logger.io.strimzi.operator.user.UserController: TRACE
+          logger.io.strimzi.operator.user.HealthCheckAndMetricsServer: DEBUG
```

Comment:
This does not work:
* These classes have no real debug or trace logs so they serve as a bad example
* This configuration changes the log level for an exiting logger. Not for a class. So it results in following error:
  ```
  Exception in thread "Log4j2-TF-1-ConfiguratonFileWatcher-8" org.apache.logging.log4j.core.config.ConfigurationException: No name attribute provided for Logger io
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.createLogger(PropertiesConfigurationBuilder.java:264)
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.build(PropertiesConfigurationBuilder.java:180)
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory.getConfiguration(PropertiesConfigurationFactory.java:56)
	at org.apache.logging.log4j.core.config.properties.PropertiesConfiguration.reconfigure(PropertiesConfiguration.java:47)
	at org.apache.logging.log4j.core.LoggerContext.onChange(LoggerContext.java:755)
	at org.apache.logging.log4j.core.util.AbstractWatcher$ReconfigurationRunnable.run(AbstractWatcher.java:93)
	at java.base/java.lang.Thread.run(Thread.java:833)
  ```

I do not know if it is possible to configure log level just for a class here. I think you would need to use something like this where you define the logger to have it working and have an example of a class that demonstrates it is working by logging at debug level regularly:

```suggestion
          logger.abstractcache.level: DEBUG
          logger.abstractcache.name: io.strimzi.operator.user.operator.cache.AbstractCache
```

or alternatively something like this to log a whole package which might make sense in some cases as well:

```
        loggers:
          logger.cache.level: DEBUG
          logger.cache.name: io.strimzi.operator.user.operator.cache
```

Defining the loggers like this makes it a bit more complicated and I wonder if we should mention also the existing loggers already defined to avoid conflicts.

This will apply to all components using Log4j2.

Reply:
I used a combination of this and Federico's suggestion, with callouts to explain the possibilities. 
I mentioned that there are default loggers, with an example for the operators. Maybe we want to list all the default loggers for each?

Reply:
I would keep default loggers out. 

Thy are convenient because it's just one line of configuration, but at the same time ambiguous. For example in the TO the config `logger.clients.level` maps to `org.apache.kafka.clients`, but it may refer to some other client package. You have to look at the config map or code to find out. Additionally, it is something we would need to keep aligned in case of changes (easy to forget).

Reply:
Thanks @fvaleri - no more updates here then